### PR TITLE
PSVAMB-35945 User ID Format of Teams Recording Ingestion

### PR DESCRIPTION
**Pull request by vberezkin-kaltura**

### CODE REVIEW

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 1]
```diff
@@ -262,8 +262,21 @@ protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
...
-		if(isset($userInfo[MicrosoftGraphFieldNames::EMAIL]))
-		{
+
+		if (isset($userInfo[MicrosoftGraphFieldNames::EMAIL])) {
+			// find user by email and return id if found
+			$filter = new KalturaUserFilter();
+			$filter->emailLike = $userInfo[MicrosoftGraphFieldNames::EMAIL];
+			$response = KBatchBase::$kClient->user->listAction($filter);
+			if ($response->totalCount > 0) {
+				$users = $response->objects;
```
Comment (hilak) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:272]: @vberezkin-kaltura you could just do a 1 line return here:

return $response->objects[0]->id

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 2]
```diff
@@ -262,8 +262,21 @@ protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
 	protected function retrieveUserId ($creatorInfo)
 	{
 		$userInfo = $creatorInfo[MicrosoftGraphFieldNames::USER];
-		if(isset($userInfo[MicrosoftGraphFieldNames::EMAIL]))
-		{
+
+		if (isset($userInfo[MicrosoftGraphFieldNames::EMAIL])) {
+			// find user by email and return id if found
+			$filter = new KalturaUserFilter();
+			$filter->emailLike = $userInfo[MicrosoftGraphFieldNames::EMAIL];
```
Comment (hilak) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:269]: @vberezkin-kaltura I think $filter->emailStartsWith is probably safer to use here.

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php (hunk 1 above)
Comment (vberezkin-kaltura) [reply plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php]: fixed

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php (hunk 2 above)
Comment (vberezkin-kaltura) [reply plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php]: fixed

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 3]
```diff
@@ -262,8 +262,17 @@ protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
 	protected function retrieveUserId ($creatorInfo)
 	{
 		$userInfo = $creatorInfo[MicrosoftGraphFieldNames::USER];
-		if(isset($userInfo[MicrosoftGraphFieldNames::EMAIL]))
-		{
+
+		if (isset($userInfo[MicrosoftGraphFieldNames::EMAIL])) {
```
Comment (hilak) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:266]: @vberezkin-kaltura the BE coding standard is to open a curly bracket on a new line.

**Comment by vberezkin-kaltura** [comment]
Successfully tested on separate instance

**Pull request merged**